File: jobs/2026-07-02__17-09-42/encore_migrate_gin_go__UnawZPq/artifacts/user/myproject/hello/hello.go
package hello

import (
	"context"
	"fmt"
)

// MessageResponse is the response shape for both endpoints.
type MessageResponse struct {
	Message string `json:"message"`
}

// GetHello returns a greeting for the given name.
//
//encore:api public path=/hello/:name method=GET
func GetHello(ctx context.Context, name string) (*MessageResponse, error) {
	return &MessageResponse{
		Message: fmt.Sprintf("Hello, %s!", name),
	}, nil
}

// GetPing returns a simple pong response.
//
//encore:api public path=/ping method=GET
func GetPing(ctx context.Context) (*MessageResponse, error) {
	return &MessageResponse{
		Message: "pong",
	}, nil
}

File: jobs/2026-07-03__05-39-37/encore_create_hello_world_ts__i92RMNe/artifacts/user/myproject/src/hello/hello.ts
import { api, Gateway } from "encore.dev/api";

// Returns a greeting for the given name.
export const get = api(
  { method: "GET", path: "/hello/:name" },
  async (params: { name: string }): Promise<{ message: string }> => {
    return { message: `Hello ${params.name}!` };
  }
);